internal/services/containers/kubernetes_nodepool.go

```diff
@@ -218,6 +218,14 @@ func SchemaDefaultNodePool() *pluginsdk.Schema {
 						Optional:     true,
 						ValidateFunc: commonids.ValidateSubnetID,
 					},
+					"pod_ip_allocation_mode": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						RequiredWith: []string{"pod_subnet_id"},
```

Comment:
Sorry for the wrong suggestion, it should be:
```suggestion
						RequiredWith: []string{"default_node_pool.0.pod_subnet_id"},
```